lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -158,6 +198,11 @@ def _verify_other_user(requesting_user, requested_username, course_key):
         Verifies whether the requesting user can access another user's view of
         the blocks in the course.
         """
+        # If accessing a public course, and requesting only content available publicly,
+        # we can allow the request.
+        if is_course_public(get_course_by_id(course_key, depth=0)) and requested_username == '':
```

Comment:
If we swap the order in this comparison, we'll save ourselves the `is_course_public` lookup if the user isn't anonymous (`requested_username != ''`).

```suggestion
        if requested_username == '' and is_course_public(course_key):
```